.cargo/config.toml

```diff
@@ -10,9 +10,26 @@ lto = true
 # https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html#lint-configuration-through-cargo
 rustflags = [
 	# rustc additional warnings:
-	"--warn",
-	"unused_crate_dependencies",
-	# clippy additional warnings:
-	"--warn",
-	"clippy::wildcard_imports",
+	"-Wunused_crate_dependencies",
+	# Rust 2018 idioms that are not yet warn-by-default:
+	"-Welided_lifetimes_in_paths",
+	"-Wunused_extern_crates",
+	"-Wexplicit_outlives_requirements",
+	# 📎 Lints that are enabled (warn/deny) by default
+	"-Wclippy::all",
+	# Restriction (optional, neutral lints)
+	"-Wclippy::dbg_macro",                       # Lint against leftover `dbg!` macros
+	"-Wclippy::todo",                            # Lint against leftover `todo!` macros
+	"-Wclippy::exit",                            # Prefer not `process::exit`ing directly
+	"-Wclippy::rest_pat_in_fully_bound_structs", # Prefer not to use `..` in fully bound structs
+	"-Wclippy::verbose_file_reads",              # Prefer simpler and more concise `fs::read_to_string`
+	# Pedantic
+	"-Wclippy::pedantic",                           # Warn about pedantic lints, except...
+	"-Aclippy::implicit_clone",                     # A lot of false positives, tuned down in Clippy bundled with Rust 1.73
+	"-Aclippy::match_same_arms",                    # It's often clearer to have the same arm twice
+	"-Aclippy::missing_errors_doc",                 # Most of our code is internal; let's not clutter the docs until...
+	"-Aclippy::missing_panics_doc",                 # ... we care about the public documentation in our shipped crates
+	"-Aclippy::module_name_repetitions",            # It seems we prefer it this way; we'd need to discuss that
```

Comment:
Definetely not blocking, but I'm happy with fixing offenses of `module_name_repetitions`. If not now, maybe we can add a note/task to the #155?

Reply:
The lint serves to reduce some stutter, e.g. `grammar::GrammarElement`, `parser_definition::TriviaParserDefinition`, `keywords::check_solidity_keywords`. In some cases it makes sense, in some it's only more correct in theory and would probably lead to some extra renamed imports or less clarity.

I'd leave it disabled for now and discuss how/if we want to proceed with it but treat it more as an internal style decision rather than in "fix lints" capacity.

Reply:
Sounds good. Thanks.